diff -aur eiskaltdcpp-2.2.9-old/CMakeLists.txt eiskaltdcpp-2.2.9/CMakeLists.txt
--- eiskaltdcpp-2.2.9-old/CMakeLists.txt	2013-08-29 22:55:02.000000000 +0530
+++ eiskaltdcpp-2.2.9/CMakeLists.txt	2013-09-26 18:43:11.191046202 +0530
@@ -151,24 +151,6 @@
 endif (FREE_SPACE_BAR_C)
 
 if (LOCAL_BOOST)
-set (boost_file_name "boost.tar.gz")
-file(GLOB BOOST_TAR_EXISTS ${PROJECT_BINARY_DIR}/${boost_file_name})
-if (NOT BOOST_TAR_EXISTS)
-file(DOWNLOAD http://sf.net/projects/eiskaltdcpp/files/Other/${boost_file_name} ${PROJECT_BINARY_DIR}/${boost_file_name})
-endif (NOT BOOST_TAR_EXISTS)
-file(GLOB BOOST_TAR_EXISTS ${PROJECT_BINARY_DIR}/${boost_file_name})
-find_program (TAR_CMD tar)
-mark_as_advanced(TAR_CMD)
-execute_process (COMMAND ${TAR_CMD} -xf ${BOOST_TAR_EXISTS}
-                 WORKING_DIRECTORY ${PROJECT_BINARY_DIR}/)
-set (Boost_INCLUDE_DIR ${PROJECT_BINARY_DIR})
-if (HAIKU)
-EXECUTE_PROCESS (
-                COMMAND patch -p0 -i ${PROJECT_SOURCE_DIR}/haiku/boost.diff
-                WORKING_DIRECTORY ${PROJECT_BINARY_DIR}/)
-endif()
-message (STATUS "USED LOCAL BOOST HEADERS")
-else (LOCAL_BOOST)
     find_package (Boost REQUIRED)
     if (Boost_VERSION VERSION_LESS 104200)
         add_definitions(-DTIMER_OLD_BOOST)
