diff -aur verlihub/configure verlihub-new/configure
--- verlihub/configure	2013-05-07 08:20:28.187164400 +0530
+++ verlihub-new/configure	2013-05-07 08:19:52.874394500 +0530
@@ -74,6 +74,8 @@
 	REAL_BUILD_DIR=`pwd`
 	# finally start cmake
 	cmake -D CMAKE_INSTALL_PREFIX=$PREFIX \
+        -D PYTHON_INCLUDE_DIR:PATH=/usr/include/python2.7 \
+        -D PYTHON_LIBRARY:FILEPATH=/usr/lib/libpython2.7.so \
         -D CMAKE_BUILD_TYPE=$BUILD_TYPE \
         -D VERBOSE=$DEBUG \
         "$SRC_DIR" \
