diff --git a/common/clean-chroot-manager32.in b/common/clean-chroot-manager32.in
index ce005e9..985e740 100644
--- a/common/clean-chroot-manager32.in
+++ b/common/clean-chroot-manager32.in
@@ -133,41 +133,60 @@ build() {
 	local mesg="Attempting to build package..."
 	echo -e "${YELLOW}---->${ALL_OFF}${BOLD} ${mesg}${ALL_OFF}"
 
-	if [[ -z "$RUNNAMCAP" ]]; then
-		makechrootpkg -c -u -r $CHROOTPATH32
-	else
-		makechrootpkg -c -u -n -r $CHROOTPATH32
+	workingcopy=$USER
+	[[ -n $SUDO_USER ]] && workingcopy=$SUDO_USER
+	[[ -z "$workingcopy" || $workingcopy = root ]] && workingcopy=copy
+
+	if [[ "$1" != "noclean" ]];then
+		option="-c"
+	fi
+
+	if [[ ! -z "$RUNNAMCAP" ]]; then
+		option="$option -n"
 	fi
 
-	# stop here is build fails
-	[[ $? -gt 0 ]] && exit 1
+	makechrootpkg $option -u -r $CHROOTPATH32 || exit 1     # stop here if build fails
 
 	# if this is the first time package has been successfully built
 	# then append the local repo to the chroot's pacman.conf
 	if [[ -z $(grep clean-chroot $CHROOTPATH32/root/etc/pacman.conf) ]]; then
-		# add a local repo to chroot
-		cat << EOF >> $CHROOTPATH32/root/etc/pacman.conf
+		# add a local repo to chroot, both in root and working copy
+		for cuser in root $workingcopy;do
+			cat << EOF >> $CHROOTPATH32/$cuser/etc/pacman.conf
 
 # Added by $PKG
 [chroot_local]
 SigLevel = Never
 Server = file:///repo
 EOF
+		done
 	fi
 
-	# setup a local repo and add adjust files in chroot
-	[[ ! -d $CHROOTPATH32/root/repo ]] && mkdir $CHROOTPATH32/root/repo
+	# source PKGBUILD to get package details
+	. $CHROOTPATH32/$workingcopy/startdir/PKGBUILD
+	# save any changes in pkgver and pkgrel to PKGBUILD
+	sed -i "s/^pkgver=.*/pkgver=$pkgver/" PKGBUILD
+	sed -i "s/^pkgrel=.*/pkgrel=$pkgrel/" PKGBUILD
+
+	# setup a local repo and add adjust files in chroot, both in root and working copy
+	[[ ! -d $CHROOTPATH32/root/repo ]] && mkdir $CHROOTPATH32/{root,$workingcopy}/repo
 	local mesg="Adding package to chroot repo..."
 	echo -e "${YELLOW}---->${ALL_OFF}${BOLD} ${mesg}${ALL_OFF}"
-	for i in *.pkg.tar.xz; do
-		cp "$i" $REPO
-		repo-add $REPO/chroot_local.db.tar.gz $REPO/"$i" || exit 1
-
-		# unsure that the chroot package matches the live pacman cache package
+	WORKINGCOPYPATH=$CHROOTPATH32/$workingcopy/repo
+	# remove earlier versions of the package from repo
+	rm -f {$REPO,$WORKINGCOPYPATH}/$pkgname*.pkg.tar.xz
+	pkgpath=$(pwd) && [[ -n "$PKGDEST" ]] && pkgpath=$PKGDEST
+	pushd $pkgpath > /dev/null
+	for i in $pkgname*-$pkgver-$pkgrel*.pkg.tar.xz; do
+		cp "$i" $REPO && repo-add $REPO/chroot_local.db.tar.gz $REPO/"$i" || exit 1
+		cp "$i" $WORKINGCOPYPATH && repo-add $WORKINGCOPYPATH/chroot_local.db.tar.gz $WORKINGCOPYPATH/"$i" > /dev/null || exit 1
+
+		# ensure that the chroot package matches the live pacman cache package
 		# which avoids checksum errors if the user builds the same $pkgname-$pkgver
 		# more than once
 		[[ -f "/var/cache/pacman/pkg/$i" ]] && rm -f "/var/cache/pacman/pkg/$i"
 	done
+	popd > /dev/null
 
 	# The rm statement above can return 1 if the file to remove is not found,
 	# causing the function to return a non-zero error code even if everything
@@ -267,7 +286,7 @@ case "$1" in
 		check2 && list
 		;;
 	n)
-		check2 && nuke
+		nuke
 		;;
 	p)
 		header && preview
@@ -279,7 +298,7 @@ case "$1" in
 			mesg=
 			create
 		fi
-		build
+		build $2
 		;;
 	u)
 		check2 && update
@@ -294,6 +313,7 @@ case "$1" in
 		echo
 		echo -e ${BOLD}" --- BUILDING ---"${ALL_OFF}
 		echo -e ${BOLD}" s) Run makepkg in clean chroot"${ALL_OFF}
+		echo -e ${BOLD}" s noclean) Run makepkg in chroot, without cleaning it first"${ALL_OFF}
 		echo
 		echo -e ${BOLD}" --- MISCELLANEOUS ---"${ALL_OFF}
 		echo -e ${BOLD}" l) List local repo in chroot (packages you build)"${ALL_OFF}
diff --git a/common/clean-chroot-manager64.in b/common/clean-chroot-manager64.in
index ac508c3..4aab6a8 100644
--- a/common/clean-chroot-manager64.in
+++ b/common/clean-chroot-manager64.in
@@ -23,7 +23,7 @@ if [[ -t 2 ]]; then
 		GREEN="${BOLD}$(tput setaf 2)"
 		RED="${BOLD}$(tput setaf 1)"
 		YELLOW="${BOLD}$(tput setaf 3)"
-	
+
 	else
 		ALL_OFF="\e[1;0m"
 		BOLD="\e[1;1m"
@@ -95,7 +95,7 @@ check() {
 			export TESTING="Enabled"
 		fi
 	fi
-	
+
 	# identify status of multilib repo
 	PACFILE="$CHROOTPATH64/root/etc/pacman.conf"
 	if [[ -f "$PACFILE" ]]; then
@@ -126,7 +126,7 @@ create() {
 			-i -e '/^#PACKAGER/ s,#,,' \
 			-i -e "/^PACKAGER/ s,=.*,=\"$PACKAGER\"," $CHROOTPATH64/root/etc/makepkg.conf
 	fi
-	
+
 	# enable pacman color output in chroot because I like it
 	sed -i -e '/^#Color/ s,#,,' $CHROOTPATH64/root/etc/pacman.conf
 }
@@ -144,42 +144,60 @@ build() {
 	local mesg="Attempting to build package..."
 	echo -e "${YELLOW}---->${ALL_OFF}${BOLD} ${mesg}${ALL_OFF}"
 
-	if [[ -z "$RUNNAMCAP" ]]; then
-		makechrootpkg -c -u -r $CHROOTPATH64
-	else
-		makechrootpkg -c -u -n -r $CHROOTPATH64
+	workingcopy=$USER
+	[[ -n $SUDO_USER ]] && workingcopy=$SUDO_USER
+	[[ -z "$workingcopy" || $workingcopy = root ]] && workingcopy=copy
+
+	if [[ "$1" != "noclean" ]];then
+		option="-c"
 	fi
 
-	# stop here is build fails
-	[[ $? -gt 0 ]] && exit 1
+	if [[ ! -z "$RUNNAMCAP" ]]; then
+		option="$option -n"
+	fi
+
+	makechrootpkg $option -u -r $CHROOTPATH64 || exit 1     # stop here if build fails
 
 	# if this is the first time package has been successfully built
 	# then append the local repo to the chroot's pacman.conf
 	if [[ -z $(grep clean-chroot $CHROOTPATH64/root/etc/pacman.conf) ]]; then
-		# add a local repo to chroot
-		cat << EOF >> $CHROOTPATH64/root/etc/pacman.conf
+		# add a local repo to chroot, both in root and working copy
+		for cuser in root $workingcopy;do
+			cat << EOF >> $CHROOTPATH64/$cuser/etc/pacman.conf
 
 # Added by $PKG
 [chroot_local]
 SigLevel = Never
 Server = file:///repo
 EOF
+		done
 	fi
-	
-	# setup a local repo and add adjust files in chroot
-	[[ ! -d $CHROOTPATH64/root/repo ]] && mkdir $CHROOTPATH64/root/repo
-	local mesg="Adding package to chroot repo..."
-	echo -e "${YELLOW}---->${ALL_OFF}${BOLD} ${mesg}${ALL_OFF}"
-	for i in *.pkg.tar.xz; do
 
-		cp "$i" $REPO
-		repo-add $REPO/chroot_local.db.tar.gz $REPO/"$i" || exit 1
+	# source PKGBUILD to get package details
+	. $CHROOTPATH64/$workingcopy/startdir/PKGBUILD
+	# save any changes in pkgver and pkgrel to PKGBUILD
+	sed -i "s/^pkgver=.*/pkgver=$pkgver/" PKGBUILD
+	sed -i "s/^pkgrel=.*/pkgrel=$pkgrel/" PKGBUILD
 
-		# unsure that the chroot package matches the live pacman cache package
+	# setup a local repo and add adjust files in chroot, both in root and working copy
+	[[ ! -d $CHROOTPATH64/root/repo ]] && mkdir $CHROOTPATH64/{root,$workingcopy}/repo
+	local mesg="Adding package to chroot repo..."
+	echo -e "${YELLOW}---->${ALL_OFF}${BOLD} ${mesg}${ALL_OFF}"
+	WORKINGCOPYPATH=$CHROOTPATH64/$workingcopy/repo
+	# remove earlier versions of the package from repo
+	rm -f {$REPO,$WORKINGCOPYPATH}/$pkgname*.pkg.tar.xz
+	pkgpath=$(pwd) && [[ -n "$PKGDEST" ]] && pkgpath=$PKGDEST
+	pushd $pkgpath > /dev/null
+	for i in $pkgname*-$pkgver-$pkgrel*.pkg.tar.xz; do
+		cp "$i" $REPO && repo-add $REPO/chroot_local.db.tar.gz $REPO/"$i" || exit 1
+		cp "$i" $WORKINGCOPYPATH && repo-add $WORKINGCOPYPATH/chroot_local.db.tar.gz $WORKINGCOPYPATH/"$i" > /dev/null || exit 1
+
+		# ensure that the chroot package matches the live pacman cache package
 		# which avoids checksum errors if the user builds the same $pkgname-$pkgver
 		# more than once
 		[[ -f "/var/cache/pacman/pkg/$i" ]] && rm -f "/var/cache/pacman/pkg/$i"
 	done
+	popd > /dev/null
 
 	# The rm statement above can return 1 if the file to remove is not found,
 	# causing the function to return a non-zero error code even if everything
@@ -311,7 +329,7 @@ case "$1" in
 		check2 && list
 		;;
 	n)
-		check2 && nuke
+		nuke
 		;;
 	p)
 		header && preview
@@ -323,7 +341,7 @@ case "$1" in
 			mesg=
 			create
 		fi
-			build
+		build $2
 		;;
 	u)
 		check2 && update
@@ -339,6 +357,7 @@ case "$1" in
 		echo
 		echo -e ${BOLD}" --- BUILDING ---"${ALL_OFF}
 		echo -e ${BOLD}" s) Run makepkg in clean chroot"${ALL_OFF}
+		echo -e ${BOLD}" s noclean) Run makepkg in chroot, without cleaning it first"${ALL_OFF}
 		echo
 		echo -e ${BOLD}" --- MISCELLANEOUS ---"${ALL_OFF}
 		echo -e ${BOLD}" l) List local repo in chroot (packages you build)"${ALL_OFF}
