diff -aur guayadeque/Trunk/CMakeLists.txt guayadeque-new/Trunk/CMakeLists.txt
--- guayadeque/Trunk/CMakeLists.txt	2015-02-01 08:11:57.574931227 +0530
+++ guayadeque-new/Trunk/CMakeLists.txt	2015-02-01 08:12:47.862446065 +0530
@@ -131,19 +131,6 @@
 # Include wxWidgets macros
 INCLUDE(${wxWidgets_USE_FILE})
 
-#
-IF( EXISTS ${PROJECT_SOURCE_DIR}/.svn )
-	# Probably a SVN workspace, determine revision level
-     	FIND_PACKAGE(Subversion)
-     	IF(Subversion_FOUND)
-        	Subversion_WC_INFO(${PROJECT_SOURCE_DIR} GUAYADEQUE)
-        	MESSAGE("Current revision is ${GUAYADEQUE_WC_REVISION}")
-		SET( _GUREVISION_ ${GUAYADEQUE_WC_REVISION})
-        ELSE(Subversion_FOUND)
-        	SET( _GUREVISION_ "" )
-	ENDIF(Subversion_FOUND)
-ENDIF( EXISTS ${PROJECT_SOURCE_DIR}/.svn ) 
-
 CONFIGURE_FILE( ${CMAKE_CURRENT_SOURCE_DIR}/src/Version.h.in ${CMAKE_CURRENT_SOURCE_DIR}/src/Version.h )
 
 #    
