diff -aur guayadeque/src/PlayerPanel.cpp guayadeque-1/src/PlayerPanel.cpp
--- guayadeque/src/PlayerPanel.cpp	2013-05-17 00:28:59.734353400 +0530
+++ guayadeque-1/src/PlayerPanel.cpp	2013-05-17 00:31:10.824394400 +0530
@@ -3073,7 +3073,8 @@
         if( m_MediaSong.m_Rating > 0 )
         {
             Body += wxT( "  " );
-            Body.Append( wxT( "★" ), m_MediaSong.m_Rating );
+            for(int i=1; i<=m_MediaSong.m_Rating; i++)
+                Body += wxT( "★" );
         }
         Body +=  wxT( "\n" ) + m_MediaSong.m_ArtistName;
         if( !m_MediaSong.m_AlbumName.IsEmpty() )
